snakemake/deployment/conda.py

```diff
@@ -365,6 +366,28 @@ def create(self, dryrun=False):
                         cmd, stderr=subprocess.STDOUT, universal_newlines=True
                     )
 
+                    # TODO run post-deploy.sh if present
+                    post_deploy_script = Path(env_file).with_suffix(".post-deploy.sh")
+                    if post_deploy_script.exists():
+                        if ON_WINDOWS:
+                            raise WorkflowError(
+                                "Post deploy script {} provided for conda env {} but unsupported on windows.".format(
+                                    post_deploy_script, env_file
+                                )
+                            )
+                        # run the script relative to current workdir
+                        logger.info(
+                            "Running post-deploy script {}...".format(
+                                post_deploy_script.relative_to(os.getcwd())
+                            )
+                        )
+                        shell.check_output(
+                            conda.shellcmd(
+                                env_path, "sh {}".format(post_deploy_script)
+                            ),
+                            stderr=subprocess.STDOUT,
+                        )
+
```

Comment:
Move this below the cleanups and outside of the if-else block (because we also need it in the if case). Further, move it into a separate method that is called from here, to avoid some code complexity in this already lengthy method.